src/flow/domain/mod.rs

```diff
@@ -987,7 +1067,8 @@ impl Domain {
                         ReplayPieceContext::Regular { .. } => {
                             debug!(self.log, "batch processed");
                         }
-                        ReplayPieceContext::Partial { for_key } => {
+                        ReplayPieceContext::Partial { for_key, ignore } => {
+                            assert!(!ignore);
```

Comment:
Actually, the more I think about it, the more I think `ignore` should be replaced with a `ReplayPieceContext` variant like `TransactionalIgnore`.